src/errors.rs

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+use std::backtrace::Backtrace;
+use std::backtrace::BacktraceStatus;
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug)]
+pub struct IronfishError {
```

Comment:
I don't see a need to have a generic error struct in this crate. If (de)serialization errors occur, `std::io::Error` is appropriate and flexible enough to convey all the information needed. If other errors need to be propagated, we can define specific error structs, following Rust best practices. I would try to keep this crate simple and with the only goal of providing multisig-related primitives, and to do that I would stick to Rust common practices